pymeasure/instruments/keithley/keithley2306.py

```diff
@@ -0,0 +1,676 @@
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    long_integration_search_enabled = Instrument.control(
+        ":SENS<ch>:LINT:SEAR?", ":SENS<ch>:LINT:SEAR %d",
+        """A boolean property that controls whether long integration search 
+        is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    long_integration_detect_enabled = Instrument.control(
+        ":SENS<ch>:LINT:DET?", ":SENS<ch>:LINT:DET %d",
+        """A boolean property that controls whether long integration detection 
+        mode is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    source_voltage = Instrument.control(
+        ":SOUR<ch>:VOLT?", ":SOUR<ch>:VOLT %g",
+        """A floating point property that controls the source voltage in volts, 
+        which takes values between 0 and 15. """,
+        validator=truncated_range,
+        values=[0, 15],
+    )
+
+    source_voltage_protection = Instrument.control(
+        ":SOUR<ch>:VOLT:PROT?", ":SOUR<ch>:VOLT:PROT %g",
+        """A floating point property that controls the source voltage protection 
+        offset in volts, which takes values between 0 and 8. """,
+        validator=truncated_range,
+        values=[0, 8],
+    )
+
+    source_voltage_protection_enabled = Instrument.measurement(
+        ":SOUR<ch>:VOLT:PROT:STAT?",
+        """A boolean property that returns the source voltage protection state. """
+    )
+
+    source_voltage_protection_clamp_enabled = Instrument.control(
+        ":SOUR<ch>:VOLT:PROT:CLAM?", ":SOUR<ch>:VOLT:PROT:CLAM %d",
+        """A boolean property that controls whether source voltage protection 
+        clamp is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    source_current_limit = Instrument.control(
+        ":SOUR<ch>:CURR?", ":SOUR<ch>:CURR %g",
+        """A floating point property that controls the source current limit in 
+        amps, which takes values between 0.006 and 5. """,
+        validator=truncated_range,
+        values=[0.006, 5],
+    )
+
+    source_current_limit_type = Instrument.control(
+        ":SOUR<ch>:CURR:TYPE?", ":SOUR<ch>:CURR:TYPE %s",
+        """A string property that controls source current limit type, which can 
+        take the values 'limit' or 'trip'. """,
+        validator=strict_discrete_set,
+        values={'limit': 'LIM', 'trip': 'TRIP'},
+        map_values=True,
+    )
+
+    source_current_limit_enabled = Instrument.measurement(
+        ":SOUR<ch>:CURR:STAT?",
+        """A boolean property that returns the source current limit state. """
+    )
+
+    last_reading = Instrument.measurement(
+        ":FETCH<ch>?",
+        """A floating point property that returns the last reading. """,
+        get_process=lambda v: float(v)
+    )
```

Comment:
In this situation, you could use the keyword parameter `cast=float` which is anyway the default, so the `get_process` parameter can be removed altogether. I must admit though that is very difficult to infer this and I know since I have seen it used in other instruments definition. Have a look at https://github.com/pymeasure/pymeasure/blob/029f8e3a6cba18730692094e64c7cbc455456b36/pymeasure/adapters/adapter.py#L75

Same comment for other similar  casting needs.

Reply:
Removed unnecessary `get_process` lambdas that do nothing more than cast to `float`.